core/trino-main/src/main/java/io/trino/operator/TopNRankingOperator.java

```diff
@@ -194,47 +207,97 @@ public TopNRankingOperator(
             outputChannelsBuilder.add(outputChannels.size());
         }
         this.outputChannels = Ints.toArray(outputChannelsBuilder.build());
+        this.partial = !generateRanking;
 
         checkArgument(maxRankingPerPartition > 0, "maxRankingPerPartition must be > 0");
 
-        if (!partitionChannels.isEmpty()) {
+        checkArgument(!maxPartialMemory.isPresent() || !generateRanking, "no partial memory on final TopN");
+        this.maxFlushableBytes = maxPartialMemory.map(dataSize -> dataSize.toBytes()).orElse(Long.MAX_VALUE);
```

Comment:
dataSize -> dataSize.toBytes()

can be replaced with:

DataSize::toBytes

Reply:
Done